src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs

```diff
@@ -0,0 +1,174 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+public class FunctionMetadataGenerator
+{
+
+    public static async Task GenerateFunctionsMetadata(string functionsPath)
+    {
+        var functionAppPath = Path.Combine(functionsPath, "function_app.py");
+
+        var functionAppContents = await File.ReadAllTextAsync(functionAppPath);
+        var functionMetadataList = ParsePythonFunctionApp(functionAppContents);
+        var functionsMetadataJson = JsonConvert.SerializeObject(functionMetadataList, Formatting.Indented);
+        var functionsMetadataPath = Path.Combine(functionsPath, "functions.metadata");
+        await File.WriteAllTextAsync(functionsMetadataPath, functionsMetadataJson);
+    }
+
+    private static List<Dictionary<string, object>> ParsePythonFunctionApp(string contents)
+    {
+        var functionMetadataList = new List<Dictionary<string, object>>();
+        var functionPattern = new Regex(@"@app\.(\w+)(?:_trigger)?\(([^)]+)\)\s+def (\w+)\(", RegexOptions.Compiled | RegexOptions.Multiline);
+        var matches = functionPattern.Matches(contents);
+
+        foreach (Match match in matches)
+        {
+            var triggerType = match.Groups[1].Value;
+            var triggerProperties = match.Groups[2].Value;
+            var functionName = match.Groups[3].Value;
+
+            var properties = ParseTriggerProperties(triggerProperties);
+
+            var bindings = new List<Dictionary<string, object>>();
+
+            switch (triggerType)
+            {
+                case "route":
+                    var authLevel = properties.GetValueOrDefault("auth_level", "Function")
+                        .Split('.')
+                        .LastOrDefault();
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", $"httpTrigger"},
+                    {"direction", "In"},
+                    {"name", "req"},
```

Comment:
Lots of repeated constants can go on top of the file for readability or a method can we created to better manage bindings for different triggerType..